src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -465,9 +465,29 @@ private static bool ParsePackageAndGenerateInstallerNodes(InstallerMetadata inst
             var newInstallers = installerMetadata.NewInstallers;
             var baseInstaller = new Installer();
             baseInstaller.InstallerUrl = url;
-            baseInstaller.InstallerSha256 = GetFileHash(path);
-            baseInstaller.Architecture = GetMachineType(path)?.ToString().ToEnumOrDefault<Architecture>() ?? Architecture.Neutral;
-
+            baseInstaller.InstallerSha256 = GetFileHash(path);
+
+            if (installerMetadata.IsZipFile)
+            {
+                baseInstaller.InstallerType = InstallerType.Zip;
+                List<string> relativeFilePaths = installerMetadata.RelativeFilePaths;
+
+                // Update target package file path to point to the first nested installer.
+                // Even for multiple nested portables, we only need to parse the first package
+                // since we already verified that the relative file paths exist in the zip archive.
+                path = Path.Combine(installerMetadata.ExtractedDirectory, relativeFilePaths.First());

```

Comment:
>First() [](http://example.com/codeflow?start=92&length=7)

This seems hacky, we are assuming all portables are same architecture, we need to check for all and if architecture mismatch, we should warn the user

Reply:
I changed the logic so that we are parsing each nested installer file and detecting the architecture. If the nested installer architectures are not all the same, I also show a nonblocking warning message to the user